lib/api/objectGet.js

```diff
@@ -36,6 +39,10 @@ function objectGet(authInfo, request, log, callback) {
                 log.debug('error processing request', { error });
                 return callback(error);
             }
+            // return ObjNotFound if the version is a delete marker
+            if (VSUtils.getdm(objMD)) {
+                return callback(errors.ObjNotFound);
```

Comment:
I think the correct error is NoSuchKey.  See s3 errors: http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html

Also, it looks like you need to add a header to the response:
"If the current version of the object is a delete marker, Amazon S3 behaves as if the object was deleted and includes x-amz-delete-marker: true in the response."